wolfcrypt/src/asn.c

```diff
@@ -40191,50 +40191,42 @@ static int ParseCRL_Extensions(DecodedCRL* dcrl, const byte* buf,
                     return ret;
                 }
                 else {
-                    if (length > 1) {
-                        int    i;
-                    #ifdef WOLFSSL_SMALL_STACK
-                        mp_int* m = (mp_int*)XMALLOC(sizeof(*m), NULL,
-                                DYNAMIC_TYPE_BIGINT);
-                        if (m == NULL) {
-                            return MEMORY_E;
-                        }
-                    #else
-                        mp_int m[1];
-                    #endif
+                #ifdef WOLFSSL_SMALL_STACK
+                    mp_int* m = (mp_int*)XMALLOC(sizeof(*m), NULL,
+                            DYNAMIC_TYPE_BIGINT);
+                    if (m == NULL) {
+                        return MEMORY_E;
+                    }
+                #else
+                    mp_int m[1];
+                #endif
```

Comment:
This is allocating the `mp_int` for its max size (`SP_INT_DIGITS`) rather than right-sizing it.  Don't you know you'll need at most `CRL_MAX_NUM_SZ * CHAR_BIT` bits here?  Looks like another opportunity for `DECL_MP_INT_SIZE_DYN()`, `NEW_MP_INT_SIZE()`, `INIT_MP_INT_SIZE()`, and `FREE_MP_INT_SIZE()`.